web/js/precallController.js

```diff
@@ -93,26 +83,28 @@
           submitForm();
         });
 
-        function hidePrecall() {
+        function hidePrecall(roomName) {
           PrecallView.hide();
           publisher && publisher.destroy();
           otNetworkTest && otNetworkTest.stopTest();
+          const username = document.getElementById('user-name-input').value.trim();
+          window.localStorage.setItem('username', username);
+          publisherOptions.name = username;
           setTimeout(() => {
             resolve({
+              roomURI: roomName,
```

Comment:
roomName variable is also declared in upper scope. Consider changing the variable name to avoid confusion.